lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/DataLoadingContext.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Common.Events;
+using Org.Apache.REEF.IO.PartitionedData;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Diagnostics;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver
+{
+    internal class DataLoadingContext<T> : IObserver<IContextStart>
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(DataLoadingContext<T>));
+
+        [Inject]
+        internal DataLoadingContext(IInputPartition<T> partition)
+        {
+            Console.WriteLine("ENTERED CONTEXT");
+        }
+
+        public void OnNext(IContextStart value)
+        {
```

Comment:
In FileSystemInputPartition which implements IInputPartition, constructor doesn't do any data downloading. Data downloading is done inside GetPartitionHandle(). 
If we choose to copy entire data from remote to local, then data downloading happens at beginning of GetPartitionHandle(). If we choose to downloading on the fly, that is what we are doing now, data will be downloaded when you iterate the IEnumerable<Row>. 
We should ensure data is downloaded with IContextStart. If data donwloading is syncronized, it will take some time for the activeContext to return to the driver, if it is asyncronized, map tasks should have some way to verify if the data is available.